editor/store/selectors.js

```diff
@@ -1202,152 +1202,280 @@ export function getNotices( state ) {
+			if ( isBoolean( list ) ) {
+				return list;
+			}
+			if ( isArray( list ) ) {
+				return includes( list, item );
+			}
+			return defaultResult;
+		};
 
-	const blockTypeIsDisabled = Array.isArray( allowedBlockTypes ) && ! includes( allowedBlockTypes, blockType.name );
-	if ( blockTypeIsDisabled ) {
-		return null;
-	}
+		const blockType = getBlockType( blockName );
+		if ( ! blockType ) {
+			return false;
+		}
 
-	if ( ! hasBlockSupport( blockType, 'inserter', true ) ) {
-		return null;
-	}
+		const { allowedBlockTypes, templateLock } = getEditorSettings( state );
 
-	return {
-		id: blockType.name,
-		name: blockType.name,
-		initialAttributes: {},
-		title: blockType.title,
-		icon: blockType.icon,
-		category: blockType.category,
-		keywords: blockType.keywords,
-		isDisabled: !! blockType.useOnce && getBlocks( state ).some( ( block ) => block.name === blockType.name ),
-	};
-}
+		const isBlockAllowedInEditor = checkAllowList( allowedBlockTypes, blockName, true );
+		if ( ! isBlockAllowedInEditor ) {
+			return false;
+		}
 
-/**
- * Given a shared block, constructs an item that appears in the inserter.
- *
- * @param {Object}           state             Global application state.
- * @param {string[]|boolean} allowedBlockTypes Allowed block types, or true/false to enable/disable all types.
- * @param {Object}           sharedBlock       Shared block, likely from getSharedBlock().
- *
- * @return {Editor.InserterItem} Item that appears in inserter.
- */
-function buildInserterItemFromSharedBlock( state, allowedBlockTypes, sharedBlock ) {
-	if ( ! allowedBlockTypes || ! sharedBlock ) {
-		return null;
-	}
+		const isEditorLocked = !! templateLock;
+		if ( isEditorLocked ) {
+			return false;
+		}
 
-	const blockTypeIsDisabled = Array.isArray( allowedBlockTypes ) && ! includes( allowedBlockTypes, 'core/block' );
-	if ( blockTypeIsDisabled ) {
-		return null;
-	}
+		const parentBlockListSettings = getBlockListSettings( state, parentUID );
+		const parentAllowedBlocks = get( parentBlockListSettings, [ 'supportedBlocks' ] );
+		const hasParentAllowedBlock = checkAllowList( parentAllowedBlocks, blockName );
+
+		const blockAllowedParentBlocks = blockType.parent;
+		const parentName = getBlockName( state, parentUID );
+		const hasBlockAllowedParent = checkAllowList( blockAllowedParentBlocks, parentName );
+
+		let isBlockAllowedInParent;
+		if ( hasParentAllowedBlock !== null && hasBlockAllowedParent !== null ) {
+			isBlockAllowedInParent = hasParentAllowedBlock || hasBlockAllowedParent;
+		} else if ( hasParentAllowedBlock !== null ) {
+			isBlockAllowedInParent = hasParentAllowedBlock;
+		} else if ( hasBlockAllowedParent !== null ) {
+			isBlockAllowedInParent = hasBlockAllowedParent;
+		} else {
+			isBlockAllowedInParent = true;
+		}
 
-	const referencedBlock = getBlock( state, sharedBlock.uid );
-	if ( ! referencedBlock ) {
-		return null;
-	}
+		if ( ! isBlockAllowedInParent ) {
```

Comment:
>Eslint says you can do:
>```js
>return isBlockAllowedInParent;
>```

This was never addressed.

I'd go further and say the `isBlockAllowedInParent` variable is unnecessary. We could just `return` at the point where it is assigned above. I don't even think an argument of it being an [explaining variable](https://refactoring.com/catalog/extractVariable.html) is valid, since it's meaning is implied as being the return value of the function (i.e. encapsulated already in the function naming).